### PR TITLE
Victor/API

### LINKED ISSUE #15: [4pt] Display missions - Lists render
- Use `useSelector()` Redux Hook to select the state slices and render lists of  missions in corresponding routes. i.e.:
```javascript
// get rockets data from the store
const rockets = useSelector(state => state.rockets);
```

- You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
- Render a table with the missions' data (as per design).

### LINKED ISSUE #19: Team3 [1pt] Create basic structure for Dragons - Setup
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create a directory for all Redux state slice files.

### LINKED ISSUE #22: [1pt] Create empty My profile - Setup (group task)
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- This view should be empty - you will add content in separate tasks.

### LINKED ISSUE #23: [1pt] Create website header with empty nav - Setup (group task)
- Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Create **empty** navigation (this will be filled in in separate tasks).

### PR DESCRIPTION
**Pull request by Victor-chiemerie**
# Highlights 🥇 

## What I have done
✅ Fetch Mission data and update the Redux store
✅ Use `useSelector()` Redux Hook to select the state missionSlice and render lists of missions in the corresponding route
✅ Render a table with the missions' data
✅ Create a reducer and action dispatcher for the "Join Mission" button
✅ Write actions and reducers for leaving missions
✅ Missions that the user has joined already should show a badge `Active Member` instead of the default `NOT A MEMBER` and a button `Leave Mission` instead of the `Join Mission` button
✅ Render a list of all joined missions (use filter())

## Why
🔴 Need to get data from API and display to DOM
🔴 Need to make UI element interactive by user

### Changes details
Please add @Victor-chiemerie in your comment so that I get notified
## Missed anything?
- [ ] Explained the purpose of this PR.
- [ ] Self-reviewed the PR.
- [ ] Added or updated test cases.
- [ ] Informed of breaking changes, testing, and migrations (if applicable).
- [ ] Updated documentation (if applicable).
- [ ] Attached are screenshots (if applicable).

## Followed Rules: 
- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
-  Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
-  Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
-  Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
-  Followed our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### CODE REVIEW

File: src/containers/Missions/Missions.jsx [hunk 1]
```diff
@@ -1,36 +1,56 @@
...
+              <td>{mission.mission_name}</td>
+              <td>{mission.description}</td>
+              <td>{mission.joined ? 'Active member' : 'Not a member'}</td>
+              <td>
+                <button
+                  type="button"
+                  className="bookButton"
+                  onClick={() => HandleClick(mission.id)}
+                >
+                  {mission.joined ? 'Leave Mission' : 'Join Mission'}
```
Comment (roniy68) [inline src/containers/Missions/Missions.jsx:39]: # Great Job 👍

File: src/containers/Missions/missionSlice.js [hunk 2]
```diff
@@ -10,21 +12,46 @@ export const missionSlice = createSlice({
...
-    }),
-    leaveMission: (state, action) => ({
+    joinMission: (state, action) => {
+      // console.log(action.payload);
+      const newState = [...state.missions];
+      const updatedMissions = newState.map((mission) => {
+        if (mission.id !== action.payload) {
+          return mission;
+        }
+        return { ...mission, joined: !mission.joined };
```
Comment (roniy68) [inline src/containers/Missions/missionSlice.js:22]: 👍  I would suggest to move the store to separate folder. I will merge this pull request and for that I will work on that

**Review by roniy68 — APPROVED** [review]
# 🔥 Status :  Approved ✅ 
---

## Great  Job @Victor-chiemerie 



There's nothing wrong with the implementation. 
### Highlights 🎖️ 
---
👍  Well done working on api 
👍  Well done on creating store 

---
## Followed Rules: 
- Made sure [no linter errors](https://github.com/microverseinc/linters-config) 🔴 
-  Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md) ✅ 
-  Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md) ✅ 
-  Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md). ✅ 
-  Followed our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md). ✅ 

### Code Readability :mag_right:
- [ ] Code is easy to read and understand. :white_check_mark:
- [ ] Code is commented where necessary. :pencil:
- [ ] Code is organized and structured. :file_folder:

### Functionality :computer:
- [ ] Code performs its intended function.  ✔️ 
- [ ] Code handles errors and edge cases appropriately. ✔️ 
- [ ] Code is optimized for performance where necessary. ✔️ 

### Testing :microscope:
- [ ] Code has been tested thoroughly. :test_tube:
- [ ] Test cases cover all possible scenarios. :heavy_check_mark:
- [ ] Test cases are well documented and easy to understand. :books:

### Security :lock:
- [ ] Code is secure and follows best practices. :shield:
- [ ] Code does not introduce vulnerabilities. :no_entry_sign:

### Documentation :page_facing_up:
- [ ] Code is well documented. :memo:
- [ ] Documentation is easy to understand. :thumbsup:
- [ ] Documentation is up to date. :calendar:

### Conclusion :tada:
- [ ] Code review has been completed. :white_check_mark:
- [ ] Any issues have been reported and discussed. :loudspeaker:
- [ ] Code is ready for merge into main branch. :rocket:


***NOTE: Please mention @roniy68 in your comments so that I can get Notification. Happy Coding :-)***
